src/checkActivities.rb

```diff
@@ -0,0 +1,143 @@
+
+        # create git instance variable
+        @git = Git.open(Dir.pwd + "/../")
+    end
+
+    # to print error
+    def self.print_error(err)
+        puts "\t#{" ERROR ".colorize(background: :red)}      #{err}"
+    end
+
+    # to print warning
+    def self.print_warning(wrn)
+        puts "\t#{" WARNING ".colorize(background: :yellow)}    #{wrn}"
+    end
+
+    # checks whether qsTr can be used in the lineStr
+    def self.qsTr_use(lineStr)
+        matchStr = /(?<qsTr>(qsTr)\()?"[\w'!?:;]+( .*)*[.,]? ?"/.match(lineStr)
+        if matchStr and not matchStr["qsTr"]
+            print_warning "#{matchStr} qsTr may not be used"
+        end
+    end
+
+    # checks copyrights updated
+    def self.activityInfo_QmlFile(lineStr, directoryName)
+        valuesToTestArray = ["author: \"Your Name &lt;yy@zz.org&gt;\"", "name: \"test/Test.qml\"", "difficulty: 1", "icon: \"test/test.svg\"",
+                             "demo: true", "title: \"Test activity\"", "description: \"\"", "goal: \"\"", "prerequisite: \"\"",
+                             "manual: \"\"", "credit: \"\"", "section: \"\""]
+
+        valuesToTestArray.each do |valueToTestStr|
+            if lineStr.include?(valueToTestStr)
+                if valueToTestStr.include?("difficulty: 1") then print_warning "#{valueToTestStr} may not be updated."
+                else print_error "#{valueToTestStr} may not be updated."
+                end
+            end
+        end
+    end
+
+    # check if directory empty
+    def self.empty_directory?
+        if @qmlFilesArray.empty? and @jsFilesArray.empty?
+            fail "No file to check! Did you enter a correct directory?"
+        end
+    end
+
+    def self.check_js_files
+        @jsFilesArray.each do |jsFile|
+
+            puts "\nFilePath: " + jsFile
+
+            File.foreach(jsFile) do |line|
+
+                if line.include?("THE GTK VERSION AUTHOR")
+                    print_error "Replace original GTK VERSION AUTHOR label by your own name"
+                end
+
+                if line.include?("THE GTK VERSION AUTHOR")
+                    print_warning "Replace original QT Quick copyright label email by your own email address"
+                end
+
+            end
+        end
+    end
+
+    def self.check_qml_files
+        @qmlFilesArray.each do |qmlFile|
+
+            puts "\nFilePath: " + qmlFile
+
+            # Assuming ActivityInfo.qml won't be much bigger
+            if File.basename(qmlFile) == "ActivityInfo.qml"
+                unless File.foreach(qmlFile).grep(/createdInVersion:/).any?
+                    print_error "ActivityInfo.qml does not have \"createdInVersion\":\n\n"
+                end
+            end
+
+            File.foreach(qmlFile) do |line|
+
+                if File.basename(qmlFile) == "ActivityInfo.qml"
+                    activityInfo_QmlFile line, @activityDirName
+                end
+
+                if line.include?("THE GTK VERSION AUTHOR")

```

Comment:
it's the same code as for js, can it be factorised?

Reply:
yes.